src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializerOptions.Converters.cs

```diff
@@ -205,15 +206,16 @@ private JsonConverter GetConverterFromAttribute(JsonConverterAttribute converter
             }
             else
             {
-                ConstructorInfo ctor = type.GetConstructor(Type.EmptyTypes);
-                if (!typeof(JsonConverter).IsAssignableFrom(type) || !ctor.IsPublic)
+                ConstructorInfo? ctor = type.GetConstructor(Type.EmptyTypes);
+                if (!typeof(JsonConverter).IsAssignableFrom(type) || ctor == null || !ctor.IsPublic)
                 {
                     ThrowHelper.ThrowInvalidOperationException_SerializationConverterOnAttributeInvalid(classTypeAttributeIsOn, propertyInfo);
```

Comment:
The exception message thrown here is the message we want in case there were no parameterless ctor, so just added that condition (`ctor == null`) within the if check above 
Message: `The converter specified on '{0}' does not derive from JsonConverter or have a public parameterless constructor.`

Reply:
Please add a test to capture this null ref bug. See https://github.com/dotnet/runtime/issues/399#issuecomment-560579442